src/lakefs_spec/client_helpers.py

```diff
@@ -189,11 +185,29 @@ def create_tag(
     try:
         return client.tags_api.create_tag(repository=repository, tag_creation=tag_creation)
     except ApiException as e:
-        if e.status == 409 and exist_ok:
-            return client.tags_api.get_tag(repository=repository, tag=tag)
+        target_commit = rev_parse(client=client, repository=repository, ref=ref)
+        existing_tag = client.tags_api.get_tag(repository=repository, tag=tag)
+        if e.status == 409 and existing_tag.commit_id == target_commit.id:
+            return existing_tag
         raise e
 
 
+def delete_tag(client: LakeFSClient, repository: str, tag: str) -> None:
```

Comment:
Should we support `Ref` or `Commit` instances as tag inputs here, like e.g. in `rev_parse`?

Reply:
Decided on just supporting `Ref` since the `Commit` object does not contain any reference to a tag. Plus, `list_tags` returns  a list of `Ref`s. Lastly, there could be edge cases such as multiple tags to a single commit (which should we then delete) and we want to avoid someone accidentally deleting someone elses tag.